components/Common/LanguageSelector/index.tsx

```diff
@@ -14,6 +15,12 @@ const dropdownStyle = {
 
 const LanguageSelector = () => {
   const [showDropdown, setShowDropdown] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useClickOutside({
+    ref,
+    handler: () => setShowDropdown(false),
+  });
```

Comment:
since `handler` has been passed in `useClickOutside` hook's `useEffect` deps, handler should be memoized or the useEffect will trigger every re-render
```suggestion
const handler = useCallback(() => setShowDropdown(false),[])

  useClickOutside({
    ref,
    handler
  });
```